taskboot/push.py

```diff
@@ -73,35 +73,41 @@ def heroku_release(target, args):
     # Load queue service
     queue = taskcluster.Queue(config.get_taskcluster_options())
 
-    # Get the list of matching artifacts as we should get only one
-    matching_artifacts = load_artifacts(args.task_id, queue, args.artifact_filter, args.exclude_filter)
-
-    # Push the Docker image
-    if len(matching_artifacts) == 0:
-        raise ValueError(f"No artifact found for {args.artifact_filter}")
-    elif len(matching_artifacts) > 1:
-        raise ValueError(f"More than one artifact found for {args.artifact_filter}: {matching_artifacts!r}")
-    else:
+    updates_payload = []
+
+    for artifact in args.artifacts:
+        heroku_dyno_type, artifact_path = artifact.split(":", 1)
+        logger.info("Searching artifact for dyno type %r with filter %r", heroku_dyno_type, artifact_path)
+
+        # Get the list of matching artifacts as we should get only one
+        matching_artifacts = load_artifacts(args.task_id, queue, artifact_path)
+
+        # Check that we only got one matching artifact
+        if len(matching_artifacts) == 0:
+            raise ValueError(f"No artifact found for {artifact_path}")
+        elif len(matching_artifacts) > 1:
+            raise ValueError(f"More than one artifact found for {artifact_path}: {matching_artifacts!r}")
+
+        # Push the Docker image
         task_id, artifact_name = matching_artifacts[0]
 
-        custom_tag_name = f"{HEROKU_REGISTRY}/{args.heroku_app}/{args.heroku_dyno_type}"
+        custom_tag_name = f"{HEROKU_REGISTRY}/{args.heroku_app}/{heroku_dyno_type}"
```

Comment:
Could you rename all occurences of `heroku_dyno_type` to `heroku_dyno_name`.
It seems [Heroku dyno type](https://devcenter.heroku.com/articles/dyno-types) is not exactly what we are using here.

Reply:
Right!